### ISSUE TITLE
[a11y] Top bar search field has no label

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#1458]
## Issue summary

The search input for top bar does not have a valid label. It relies instead on a `placeholder` value, which may not be conveyed consistently to assistive technology users.

## Expected behavior

All `<input>` elements will be tied programmatically to a `<label>' with a valid `for`/`id` pair to consistently convey its purpose to users. [See the WCAG technique for labeling fields.](https://www.w3.org/TR/WCAG20-TECHS/H44.html)

## Actual behavior

The field has no label.

### React

```jsx
const searchFieldMarkup = (
      <TopBar.SearchField
        onChange={handleSearchChange}
        value={searchText}
        placeholder="Search"
      />
    );
```

### HTML output

```
<input class="Polaris-TopBar-SearchField__Input" placeholder="Search" type="search" autocapitalize="off" autocomplete="off" autocorrect="off" value="">
```

## Steps to reproduce the problem

There are a couple of ways to find this problem.

### Test in the browser output

1. Navigate to the top bar page: https://polaris.shopify.com/components/structure/top-bar
1. View the HTML output of the text input for any example

### Use an automated tool

1. Open the code sandbox in its own tab in Chrome: https://38rxl3ok4m.codesandbox.io/
1. Run the [accessibility audit built into Chrome](https://developers.google.com/web/tools/lighthouse/)

<img width="908" alt="Screenshot of invalid labeling results in Lighthouse" src="https://user-images.githubusercontent.com/1462085/57557995-e9399400-7330-11e9-82c1-87410726553e.png">

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.1.4
- Browser: All
- Device: All
- Operating System: All
- WCAG 2.1 Success Criteria: [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships) (Level A), [3.3.2 Labels or Instructions](https://www.w3.org/TR/WCAG21/#labels-or-instructions) (Level A)

**Comment by marschattha** [comment]
Can I work on this as a first issue?

**Comment by alex-page** [comment]
@marschattha This should be resolved in the #1565 PR. Please feel free to jump on any time you notice an issue with the label "Good first issue".
